.github/workflows/release.yaml

```diff
@@ -0,0 +1,44 @@
+name: release
+
+on:
+  push:
+    branches:
+      - 'user/**'
+      - 'feature/**'
+      - 'improvement/**'
+      - 'bugfix/**'
+      - 'dependabot/**'
+      - 'w/**'
+      - 'q/**'
```

Comment:
You can change the on trigger to either a workflow dispatch or release/tag. You can leave push on branches but in this case the tag on the docker image should something like `${{ github.sha }}`

Anyway the goal is that we are able to trigger this build when tagging the repo or forcing it (workflow dispatch) and let the workflow tag the repo itself along with pushing an image that matches the tag name example: `3.19.0`